lib/utils.nix

```diff
@@ -2,33 +2,28 @@
 
 {
   config.lib.stylix = {
-    # get the opacity values as strings in hexadecimal, integer and floating point values for the four opacity options
-    desktopOpacity-hex = lib.toHexString ((((builtins.ceil (config.stylix.opacity.desktop * 100)) * 255) / 100));
-    desktopOpacity-int = builtins.toString (builtins.ceil (config.stylix.opacity.desktop * 100));
-    desktopOpacity-float = builtins.toString config.stylix.opacity.desktop;
+    # get the opacity values as strings in hexadecimal, integer and floating point values
 
+    opacityToHex = opacity: lib.toHexString (
+      ((builtins.ceil (opacity * 100)) * 255 / 100)
+    );
 
-    applicationsOpacity-hex = lib.toHexString ((((builtins.ceil (config.stylix.opacity.applications * 100)) * 255) / 100));
-    applicationsOpacity-int = builtins.toString (builtins.ceil (config.stylix.opacity.applications * 100));
-    applicationsOpacity-float = builtins.toString config.stylix.opacity.applications;
+    opacityToInt = opacity: builtins.toString (
+      builtins.ceil (opacity * 100));
 
+    opacityToFloat = opacity: builtins.toString
+      opacity;
 
-    terminalOpacity-hex = lib.toHexString ((((builtins.ceil (config.stylix.opacity.terminal * 100)) * 255) / 100));
-    terminalOpacity-int = builtins.toString (builtins.ceil (config.stylix.opacity.terminal * 100));
-    terminalOpacity-float = builtins.toString config.stylix.opacity.terminal;
-
-
-    popupsOpacity-hex = lib.toHexString ((((builtins.ceil (config.stylix.opacity.popups * 100)) * 255) / 100));
-    popupsOpacity-int = builtins.toString (builtins.ceil (config.stylix.opacity.popups * 100));
-    popupsOpacity-float = builtins.toString config.stylix.opacity.popups;
-
-    backgroundPolarity =
+    polarityFrom = colors:
```

Comment:
Currently the function takes the `color` attr set.
This makes it relatively simple to call the function.

However this argument is not type checked. This will result in misleading errors at best and is documented nowhere atm. (except for this comment...)

Should we take `red`, `green` and `blue` as arguments instead?